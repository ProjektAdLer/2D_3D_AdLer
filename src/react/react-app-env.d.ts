/// <reference types="react-scripts" />

// Images
declare module '*.jpg';
declare module '*.png';
declare module '*.env';

// 3D types
declare module '*.glb';
declare module '*.stl';

declare module '*.jpeg';
declare module '*.gif';
declare module '*.mp4';
