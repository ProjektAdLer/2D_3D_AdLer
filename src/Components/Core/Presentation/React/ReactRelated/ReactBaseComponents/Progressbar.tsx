import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";
import tailwindMerge from "../../../Utils/TailwindMerge";

type ProgressbarProps = {
  /* Progress bar arguments */
  value: number;
  tooltip?: string;
  max: number;
  progressbarText?: string; // if no displaytext is defined progressbar displays value
  barClassName?: string;
  /* icon arguments */
  button: boolean;
  icon?: string;
  iconClassName?: string;
  iconTextClassName?: string;
  iconText?: string; // text will be displayed in icon. If no icon is defined then level will not be displayed
};

export default function Progressbar(props: Readonly<ProgressbarProps>) {
  return (
    <div className="flex items-center text-center" title={props.tooltip}>
      {/* icon */}
      {props.icon && (
        <div className="relative items-center align-middle">
          {/* button */}
          {props.button === true && (
            <StyledButton>
              <img
                className={tailwindMerge(
                  props.iconClassName,
                  "absolute align-middle items-center w-[48px] lg:w-[69px]",
                )}
                src={props.icon}
                alt="Temp Icon"
              />
              <div
                className={tailwindMerge("z-10", props.iconTextClassName ?? "")}
              >
                {props.iconText}
              </div>
            </StyledButton>
          )}
          {/* no button */}
          {props.button === false && (
            <div className="relative flex items-center justify-center w-10 h-10 p-1 aspect-square mobile-landscape:w-9 mobile-landscape:h-9 sm:w-10 sm:h-10 md:w-14 md:h-14 lg:w-16 lg:h-16">
              <img
                className={tailwindMerge(
                  props.iconClassName,
                  "absolute align-middle items-center w-[48px] lg:w-[69px]",
                )}
                src={props.icon}
                alt="Temp Icon"
              />
              <div
                className={tailwindMerge("z-10", props.iconTextClassName ?? "")}
              >
                {props.iconText}
              </div>
            </div>
          )}
        </div>
      )}
      {/* bar */}
      {/* progress bar */}
      <div
        className={tailwindMerge(
          props.barClassName,
          "relative align-middle rounded-full bg-adlerblue/80 dark:bg-gray-700 -z-20",
        )}
      >
        <div
          className="absolute h-full rounded-full bg-adlerdarkblue -z-10"
          style={{
            width:
              Math.min((props.value / props.max) * 100, 100).toString() + "%",
          }}
        ></div>
        {props.progressbarText ?? Math.min(props.value, props.max)}
      </div>
    </div>
  );
}
