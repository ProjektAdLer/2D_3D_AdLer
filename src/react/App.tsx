import React from 'react';
import './App.css';
import BabylonCanvas from './components/BabylonCanvas';

function App() {
	return (
		<React.Fragment>
			<BabylonCanvas style={{ width: '100%', position: 'absolute' }} />
		</React.Fragment>
	);
}

export default App;
