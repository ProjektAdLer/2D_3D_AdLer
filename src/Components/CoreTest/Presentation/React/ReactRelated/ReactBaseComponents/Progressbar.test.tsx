import { render } from "@testing-library/react";
import Progressbar from "../../../../../Core/Presentation/React/ReactRelated/ReactBaseComponents/Progressbar";

describe("Progressbar", () => {
  test("should render", () => {
    const componentUnderTest = render(<Progressbar value={1} max={10} />);
    expect(componentUnderTest.container).toMatchSnapshot();
  });
});
