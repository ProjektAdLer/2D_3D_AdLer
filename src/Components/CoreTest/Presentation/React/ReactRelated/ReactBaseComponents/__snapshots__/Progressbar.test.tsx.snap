// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progressbar should render 1`] = `
<div>
  <div
    class="flex items-center text-center"
  >
    <div
      class="relative -z-20 rounded-full bg-adlerblue/80 align-middle dark:bg-gray-700"
    >
      <div
        class="absolute -z-10 h-full rounded-full bg-adlerdarkblue"
        style="width: 10%;"
      />
      1
    </div>
  </div>
</div>
`;
